src/readbin/ida.ml

```diff
@@ -14,7 +16,10 @@ let run cmd =
   let r = In_channel.input_lines inp in
   In_channel.close inp; r
 
-let system cmd = Unix.system cmd |> ignore
+let system cmd =
+  let return = Sys.command cmd in
```

Comment:
1. `return` is a verb, usually we use `r` or `result` for the result of call
2. never write `then ()` this is as bad as writing `if x then false else true`, write:
```
if Sys.command cmd <> 0 then raise (Cmderr cmd);
```
3. Exception, exported out of the module is like big intermodular global variable, it must have big and shiny name, like `Command_failed`

And now the big issue. You silently changed the behavior and semantics of the library function, and didn't fix all the code that uses it. From the observer perspective, `get_symbols` was a function that doesn't fail, returning empty set in case if it is impossible to get the symbols from the resource. Now it fails instead. As a result you broke our main `bap` utility.